src/OrchardCore.Modules/OrchardCore.Media/Services/ChunkFileUploadService.cs

```diff
@@ -200,9 +200,14 @@ private static FileStream CreateTemporaryFile(string tempPath, long size)
 
     private static string CalculateHash(params string[] parts)
     {
-        var hash = SHA256.HashData(Encoding.UTF8.GetBytes(string.Join(string.Empty, parts)));
+        var hash = new XxHash64();
 
-        return Convert.ToHexString(hash);
+        foreach (var part in parts)
+        {
+            hash.Append(Encoding.UTF8.GetBytes(part));
```

Comment:
No string concatenation necessary with this one.

Reply:
Note: since you're changing the algorithm, I presume the result is only used internally and can change at any time.
If so, you may want to use the UTF-16 representation here, to avoid an intermediate `string` allocation.

Something like `hash.Append(MemoryMarshal.AsBytes<char>(part));`

Reply:
Here is a benchmark with UTF8 vs Marshal.AsBytes, with and without the HEX encoding. Was wondering is the encoding would cancel out the benefits, still good. A little slower however, probably because of the extra hashing work.

```
| Method       | Mean      | Error     | StdDev   | Gen0   | Allocated |
|------------- |----------:|----------:|---------:|-------:|----------:|
| UTF8         |  77.37 ns | 39.491 ns | 2.165 ns | 0.0348 |     328 B |
| UTF8ToHex    |  82.21 ns | 51.756 ns | 2.837 ns | 0.0408 |     384 B |
| Marshal      | 104.60 ns | 25.270 ns | 1.385 ns | 0.0196 |     184 B |
| MarshalToHex |  91.35 ns |  7.423 ns | 0.407 ns | 0.0254 |     240 B |
```